examples/knative/python/kn-py-vro/handler.py

```diff
@@ -0,0 +1,270 @@
+                    }
+                }
+        if not 'type' in param: # if no vc managed object was found, pass the data as JSON
+            param['type'] = 'string'
+            param['value'] = {"string": {"value": json.dumps(value)}}
+
+    elif type(value) == list:
+        param['type'] = "Properties"
+        properties = []
+        for v in value:
+            if 'Key' in v and 'Value' in v:
+                # pass the key and value to this function recursively to get a correctly structured value
+                vroParam = getVroInputParam((v["Key"], v["Value"]))
+                properties.append({"key": v["Key"], "value": vroParam["value"]})
+        param['value'] = {"properties": {"property": properties}}
+
+    elif value is None:
+        param = None
+
+    else:  # Unhandled data type - try and wrangle to a string
+        app.logger.debug(f'{bgc.WARNING}Unhandled data type "{type(value).__name__}" - forcing to string{bgc.ENDC}')
+        param['type'] = "string"
+        param['value'] = {"string": {"value": str(value)}}
+
+    if type(param) == dict:  # set the parameter name for valid types
+        param["name"] = name[0].lower() + name[1:]
+    return param
+
+
+def get_response(message, code):
+        app.logger.error(message)
+        msg = {
+            'status': code,
+            'error': message,
+        }
+        resp = jsonify(msg)
+        resp.status_code = code
+        return resp
+
+
+app = Flask(__name__)
+@app.route('/', methods=['POST'])
+def handle():
+    """
+    Handles a request to the function
+
+    """
+    global vchost
+
+    # Load the Events that function gets from vCenter through the Event Router
+    app.logger.debug(f'{bgc.HEADER}Reading Cloud Event: {bgc.ENDC}')
+    cevent = from_http(request.headers, request.get_data(),None)
+
+    if cevent._attributes.get("datacontenttype").lower() != "application/json":
+            return get_response(f'invalid datacontenttype for cloud event: {cevent._attributes.get("datacontenttype")}', 500)
+    app.logger.debug(f'{bgc.OKBLUE}Event > {bgc.ENDC}{type(cevent)} - {cevent}')
+
+    try:
+        # CloudEvent - simple validation
+        source = cevent._attributes['source']
+        event = cevent.data
+        _ = event['Key']
+    except KeyError as err:
+        traceback.print_exc(limit=1, file=sys.stderr)  # providing traceback since it helps debug the exact key that failed
+        return get_response(f'Invalid JSON, required key not found > KeyError: {err} > {cevent}', 500)
+    except TypeError as err:
+        traceback.print_exc(limit=1, file=sys.stderr)  # providing traceback since it helps debug the exact key that failed
+        return get_response(f'Invalid JSON, data not iterable > TypeError: {err}', 500)
+
+    app.logger.debug(f'{bgc.HEADER}Validating Input data: {bgc.ENDC}')
+    app.logger.debug(f'{bgc.OKBLUE}Event > {bgc.ENDC}{json.dumps(event, indent=4, sort_keys=True)}')
```

Comment:
question: do we want to dump via json or just use the object's string representation? (I think the class implements str IIRC)

Reply:
That is correct, although in terms of readability in the log output, the string representation of CloudEvent is an unformatted single line, whereas this line makes it easier to read. TBH this is probably only useful during development of the function, so I'm happy either way.

Reply:
Ok, if the `json.dumps` output does not mess up the object/content and makes it easier to read, go for it!